test/test_common/logging.h

```diff
@@ -61,21 +61,21 @@ class LogRecordingSink : public Logger::SinkDelegate {
   std::vector<std::string> messages_;
 };
 
-typedef std::vector<std::pair<std::string, std::string>> ExpectedLoggingPairs;
+typedef std::vector<std::pair<std::string, std::string>> ExpectedLogSequence;
 
-// Validates that when the stmt is executed, a sequence of emmitted logs matches a sequence of
-// expected log messages and log level pairs (ExpectedLoggingPairs). Note that both sequences must
+// Validates that when the stmt is executed, a sequence of emmitted log messages matches the
+// sequence of expected logs/log level pair (ExpectedLogSequence). Note that both sequences must
 // respect the same order.
-#define EXPECT_LOG_SEQ(expected_logging_pairs, stmt)                                               \
+#define EXPECT_LOG_SEQ(expected_log_sequence, stmt)                                                \
   do {                                                                                             \
     LogLevelSetter save_levels(spdlog::level::trace);                                              \
     LogRecordingSink log_recorder(Logger::Registry::getSink());                                    \
     stmt;                                                                                          \
-    ASSERT_EQ(expected_logging_pairs.size(), log_recorder.messages().size());                      \
-    auto expected_it = expected_logging_pairs.begin();                                             \
+    ASSERT_EQ(expected_log_sequence.size(), log_recorder.messages().size());                       \
+    auto expected_it = expected_log_sequence.begin();                                              \
     auto actual_it = log_recorder.messages().begin();                                              \
-    for (; expected_it != expected_logging_pairs.end() &&                                          \
-           actual_it != log_recorder.messages().end();                                             \
+    for (;                                                                                         \
+         expected_it != expected_log_sequence.end() && actual_it != log_recorder.messages().end(); \
```

Comment:
you've already asserted that the sizes are the same, so no need to check both iterators for end.  Might be cleaner to just index-loop over the parallel vectors and index-ref the vectors.

Also -- I can't comment on the lines below you didn't modify, but it might be better to redefine EXPECT_LOG_CONTAINS in terms of EXPECT_LOG_SEQ as a special case, to avoid having my very kludgy log-parsing code in there twice :)

```
#define EXPECT_LOG_CONTAINS(loglevel, substr, stmt) EXPECT_LOG_SEQ(ExpectedLogSequence{{loglevel, substr}, stmt)
```

I'm not sure if that will actually work as I wrote it :)